static/src/javascripts/projects/membership/digitalpack-tab.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+import $ from 'lib/$';
+import fetch from 'lib/fetch';
+import config from 'lib/config';
+import reportError from 'lib/report-error';
+import formatters from 'membership/formatters';
+import stripe from 'membership/stripe';
+
+const PACKAGE_COST = '.js-dig-package-cost';
+const PAYMENT_FORM = '.js-dig-card-details';
+const SUBSCRIBER_ID = '.js-dig-subscriber-id';
+const REMAINING_TRIAL_LENGTH = '.js-dig-remaining-trial-length';
+const REMAINING_TRIAL_LENGTH_CONTAINER =
+    '.js-dig-remaining-trial-length-container';
+const PACKAGE_CURRENT_RENEWAL_DATE = '.js-dig-current-renewal-date';
+const PACKAGE_CURRENT_PERIOD_END = '.js-dig-current-period-end';
+const PACKAGE_CURRENT_PERIOD_START = '.js-dig-current-period-start';
+const PACKAGE_NEXT_PAYMENT_DATE = '.js-dig-next-payment-date';
+const PACKAGE_NEXT_PAYMENT_PRICE = '.js-dig-next-payment-price';
+const PACKAGE_NEXT_PAYMENT_CONTAINER = '.js-dig-next-payment-container';
+const PACKAGE_INTERVAL = '.js-dig-plan-interval';
+const DETAILS_JOIN_DATE = '.js-dig-join-date';
+const NOTIFICATION_CANCEL = '.js-dig-cancel-tier';
+const DIGITALPACK_DETAILS = '.js-dig-details';
+const DIGITALPACK_PRODUCT = '.js-dig-product';
+const UP_SELL = '.js-dig-up-sell';
+const DIG_INFO = '.js-dig-info';
+const LOADER = '.js-dig-loader';
+const IS_HIDDEN_CLASSNAME = 'is-hidden';
+const ERROR = '.js-dig-error';
+
+const hideLoader = () => {
+    $(LOADER).addClass(IS_HIDDEN_CLASSNAME);
+};
+
+/**
+ * @param {{
+ *  optIn,
+ *  joinDate,subscriberId,
+ *  subscription: {trialLength, nextPaymentDate,nextPaymentPrice,renewalDate,
+ *  plan: {name,amount,interval}}
+ * }} userDetails
+ */
+const populateUserDetails = userDetails => {
```

Comment:
It would be pretty cool if you could encode the above JSDoc-style comment into a Flow type!

Reply:
This would be a common type between this and membership-tab, where should it live?

Reply:
Good question. Maybe create a `membership.js` in our custom [Flow types folder](https://github.com/guardian/frontend/tree/master/static/src/javascripts/__flow__/types)